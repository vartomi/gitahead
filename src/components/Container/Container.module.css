.container {
    margin: 32px;
    display: flex;
    flex-direction: column;
    align-items: center;
    flex-grow: 1;
}