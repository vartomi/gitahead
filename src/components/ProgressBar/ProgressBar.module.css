.container {
    text-align: center;
}

.container.compact {
    position: absolute;
    top: 12px;
    right: 8px;
}

.animation {
    -webkit-animation:spin 3s linear infinite;
    -moz-animation:spin 3s linear infinite;
    animation:spin 3s linear infinite;
}

@-moz-keyframes spin { 100% { -moz-transform: rotate(360deg); } }
@-webkit-keyframes spin { 100% { -webkit-transform: rotate(360deg); } }
@keyframes spin { 100% { -webkit-transform: rotate(360deg); transform:rotate(360deg); } }