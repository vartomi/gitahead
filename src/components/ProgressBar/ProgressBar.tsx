import React from 'react';
import styles from './ProgressBar.module.css';
import { ReactComponent as Logo } from '../../assets/logo.svg';
import classNames from 'classnames';

type ProgressBarProps = {
    compact?: boolean;
}

const ProgressBar: React.FC<ProgressBarProps> = ({ compact }) => {
    const size: number = compact ? 24 : 64;
    return (
        <div className={classNames(styles.container, { [styles.compact]: compact })}>
            <Logo height={size} width={size} className={styles.animation} />
            {!compact && <div>Loading...</div>}
        </div>
    );
}

export default ProgressBar;
