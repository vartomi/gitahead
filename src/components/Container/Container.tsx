import React, { useEffect, useState } from 'react';
import styles from './Container.module.css';
import Search from '../Search/Search';
import { User, getUsers } from '../../services/users';
import ProgressBar from '../ProgressBar/ProgressBar';
import UserProfile from '../UserProfile/UserProfile';

const Container: React.FC = () => {
    const [users, setUsers]: [User[] | null, any] = useState(null);
    const [selectedUser, setSelectedUser]: [User | null, any] = useState(null);

    useEffect(() => {
        getUsers().then(setUsers);
    }, []);

    return (
        <main className={styles.container}>
            {users ? <Search onSelect={setSelectedUser} /> : <ProgressBar />}
            {selectedUser && <UserProfile user={selectedUser} />}
        </main>
    );
}

export default Container;
