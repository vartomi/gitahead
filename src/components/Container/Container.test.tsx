import { shallow } from 'enzyme'
import Container from './Container';
import React from 'react';
import ProgressBar from '../ProgressBar/ProgressBar';

describe('Container component', () => {
    it('should be rendered with ProgressBar', () => {
        const wrapper = shallow(<Container />)
        expect(wrapper.find(ProgressBar).length).toBeTruthy();
    });

    it('should render snapshot', () => {
        const wrapper = shallow(<Container />)
        expect(wrapper).toMatchSnapshot();
    });
});