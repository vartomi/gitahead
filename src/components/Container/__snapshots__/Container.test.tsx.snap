// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render snapshot 1`] = `
<main
  className="container"
>
  <ProgressBar />
</main>
`;
